src/wp-includes/class-wp-block-patterns-registry.php

```diff
@@ -65,6 +68,21 @@ final class WP_Block_Patterns_Registry {
 	 * @return bool True if the pattern was registered with success and false otherwise.
 	 */
 	public function register( $pattern_name, $pattern_properties ) {
+		global $wp_current_filter;
+
+		if ( ! empty( $wp_current_filter ) && ! in_array( 'init', $wp_current_filter, true ) ) {
```

Comment:
The goal of this fix is to ensure this method is only called when `init` is currently firing. [`doing_action()` does this](https://developer.wordpress.org/reference/functions/doing_action/):

```suggestion
		if ( ! doing_action( 'init' ) ) {
```